tools/consumer_offsets_recovery/main.py

```diff
@@ -0,0 +1,163 @@
+import os
+from pathlib import Path
+
+import subprocess
+from kafka import KafkaAdminClient
+from kafka import KafkaConsumer
+from kafka.admin.new_topic import NewTopic
+from kafka import TopicPartition
+from kafka import OffsetAndMetadata
+from jproperties import Properties
+import logging
+
+logger = logging.getLogger('cg-recovery-tool')
+
+
+def read_offsets(admin: KafkaAdminClient):
+
+    groups_dict = {}
+    groups = admin.list_consumer_groups()
+    for g, _ in groups:
+        logger.info(f"reading group '{g}' offsets")
+        offsets = admin.list_consumer_group_offsets(group_id=g)
+        groups_dict[g] = offsets
+
+    return groups_dict
+
+
+def read_config(path):
+    logger.debug(f"reading configuration from {path}")
+    cfg = Properties()
+    with open(path, 'rb') as f:
+        cfg.load(f)
+    cfg_dict = {}
+    for k, v in cfg.items():
+        cfg_dict[k] = v.data
+
+    return cfg_dict
+
+
+def seek_to_file(path, cfg, dry_run):
+
+    logger.debug(f"reading offsets file: {path}")
+    offsets = {}
+    with open(path, 'r') as f:
+        group = ""
+        consumer = None
+        for i, l in enumerate(f):
+            if i == 0:
+                group = l.strip()
+                if not dry_run:
+                    consumer = KafkaConsumer(group_id=group, **cfg)
```

Comment:
Nit: No need to check if i is 0 within the loop, can just initialize consumer before `for` loop and perform some error checking against the value of the first line in the file.